src/applications/spatial.c

```diff
@@ -314,340 +477,382 @@ int starsh_ssdata_generate_el(STARSH_ssdata **data, STARSH_int count, ...)
- * @ingroup app-spatial
- * */
+    //! Free memory of @ref STARSH_ssdata object.
+    /*! @sa starsh_ssdata_new(), starsh_ssdata_init(), starsh_ssdata_generate().
+     * @ingroup app-spatial
+     * */
 {
     starsh_particles_free(&data->particles);
 }
 
 static int starsh_ssdata_get_kernel_1d(STARSH_kernel **kernel,
-        enum STARSH_SPATIAL_KERNEL type)
-// Get kernel for 1-dimensional spatial statistics problem.
-// This function is static not to be visible outside this module.
+	enum STARSH_SPATIAL_KERNEL type)
+    // Get kernel for 1-dimensional spatial statistics problem.
+    // This function is static not to be visible outside this module.
 {
     switch(type)
     {
-        case STARSH_SPATIAL_EXP:
-            *kernel = starsh_ssdata_block_exp_kernel_1d;
-            break;
-        case STARSH_SPATIAL_EXP_SIMD:
-            *kernel = starsh_ssdata_block_exp_kernel_1d_simd;
-            break;
-        case STARSH_SPATIAL_SQREXP:
-            *kernel = starsh_ssdata_block_sqrexp_kernel_1d;
-            break;
-        case STARSH_SPATIAL_SQREXP_SIMD:
-            *kernel = starsh_ssdata_block_sqrexp_kernel_1d_simd;
-            break;
+	case STARSH_SPATIAL_EXP:
+	    *kernel = starsh_ssdata_block_exp_kernel_1d;
+	    break;
+	case STARSH_SPATIAL_EXP_SIMD:
+	    *kernel = starsh_ssdata_block_exp_kernel_1d_simd;
+	    break;
+	case STARSH_SPATIAL_SQREXP:
+	    *kernel = starsh_ssdata_block_sqrexp_kernel_1d;
+	    break;
+	case STARSH_SPATIAL_SQREXP_SIMD:
+	    *kernel = starsh_ssdata_block_sqrexp_kernel_1d_simd;
+	    break;
 #ifdef GSL
-        case STARSH_SPATIAL_MATERN:
-            *kernel = starsh_ssdata_block_matern_kernel_1d;
-            break;
-        case STARSH_SPATIAL_MATERN_SIMD:
-            *kernel = starsh_ssdata_block_matern_kernel_1d_simd;
-            break;
-        case STARSH_SPATIAL_MATERN2:
-            *kernel = starsh_ssdata_block_matern2_kernel_1d;
-            break;
-        case STARSH_SPATIAL_MATERN2_SIMD:
-            *kernel = starsh_ssdata_block_matern2_kernel_1d_simd;
-            break;
+	case STARSH_SPATIAL_MATERN:
+	    *kernel = starsh_ssdata_block_matern_kernel_1d;
+	    break;
+	case STARSH_SPATIAL_MATERN_SIMD:
+	    *kernel = starsh_ssdata_block_matern_kernel_1d_simd;
+	    break;
+	case STARSH_SPATIAL_MATERN2:
+	    *kernel = starsh_ssdata_block_matern2_kernel_1d;
+	    break;
+	case STARSH_SPATIAL_MATERN2_SIMD:
+	    *kernel = starsh_ssdata_block_matern2_kernel_1d_simd;
+	    break;
 #else
-        case STARSH_SPATIAL_MATERN:
-        case STARSH_SPATIAL_MATERN_SIMD:
-        case STARSH_SPATIAL_MATERN2:
-        case STARSH_SPATIAL_MATERN2_SIMD:
-            STARSH_ERROR("Matern kernel requires GSL library, which was "
-                    "not found");
-            return STARSH_WRONG_PARAMETER;
+	case STARSH_SPATIAL_MATERN:
+	case STARSH_SPATIAL_MATERN_SIMD:
+	case STARSH_SPATIAL_MATERN2:
+	case STARSH_SPATIAL_MATERN2_SIMD:
+	case STARSH_SPATIAL_PARSIMONIOUS_SIMD:
```

Comment:
case STARSH_SPATIAL_PARSIMONIOUS shall return error here, since this kernel does not support generic value of dimensionality (ndim=1, although kernel is meant only for 2d case).